### PR TITLE
cmd: Allow `add-package` to select version of package

### LINKED ISSUE #6549: "caddy add-package" could  allow adding by version
At the moment we cant control the version of the package added ?

The golang module system could allow us to add by tag or branch. 

I expect many developers would want to use a tag ( as in git tag ) to help remediate supply channel attacks or other things of this nature.

```sh
caddy add-package -h

Downloads an updated Caddy binary with the specified packages (module/plugin)
added. Retains existing packages. Returns an error if the any of packages are
already included. EXPERIMENTAL: May be changed or removed.

Usage:
  caddy add-package <packages...> [flags]

Flags:
  -h, --help          help for add-package
  -k, --keep-backup   Keep the backed up binary, instead of deleting it
```

This would bring it inline with the ```list-modules --packages --versions``` command, which is version aware, so that its all consistent  and all commands are version aware.

```sh
list-modules --packages --versions

Standard modules: 121

exec v0.0.0-20240603212820-a42a5b2ae10f github.com/abiosoft/caddy-exec
http.handlers.exec v0.0.0-20240603212820-a42a5b2ae10f github.com/abiosoft/caddy-exec

  Non-standard modules: 2
```

There is no point making ```caddy remove-package ```version aware since the binary can only have one version of a package.

## final look and feel

For example, its is currently:

```

caddy add-package -h

Downloads an updated Caddy binary with the specified packages (module/plugin)
added. Retains existing packages. 

Returns an error if the any of packages are
already included. EXPERIMENTAL: May be changed or removed. 
```

with versioning it becomes :

```

caddy add-package -h

Downloads an updated Caddy binary with the specified packages (module/plugin)
added. Retains existing packages. 

Returns a success if the any of package versions are changed that are
already included. EXPERIMENTAL: May be changed or removed. 


```

### PR DESCRIPTION
**Pull request by Botelho31**
Closes #6549

### CODE REVIEW

File: cmd/packagesfuncs.go [hunk 1]
```diff
@@ -46,6 +46,25 @@ func cmdUpgrade(fl Flags) (int, error) {
 	return upgradeBuild(pluginPkgs, fl)
 }
 
+func splitModule(arg string) (module, version string, err error) {
+	const versionSplit = "@"
+
+	// accommodate module paths that have @ in them, but we can only tolerate that if there's also
+	// a version, otherwise we don't know if it's a version separator or part of the file path (see #109)
```
Comment (mholt) [inline cmd/packagesfuncs.go:53]: What is `#109`? #109 ??

**Review by mholt — CHANGES_REQUESTED** [review]
Thank you -- LGTM, although I am puzzled by one of the comments.

File: cmd/packagesfuncs.go (hunk 1 above)
Comment (Botelho31) [reply cmd/packagesfuncs.go:53]: This is actually a issue ID from xcaddy as i copied a large chunk of the split module function from there. I guess i missed this, ill remove it

Comment (mholt) [reply cmd/packagesfuncs.go]: Ah, haha, I thought it looked familiar. Thanks

**Review by mholt — APPROVED** [review]

**Pull request merged**